### PR TITLE
rule: Rule.Type to allow adding/listing unreachable (RTN_UNREACHABLE)…

### PR DESCRIPTION
**Pull request by oxtoacart**
… rules

Updates #710

### CODE REVIEW

File: rule.go [hunk 1]
```diff
@@ -28,6 +28,12 @@ type Rule struct {
 	IPProto           int
 	UIDRange          *RuleUIDRange
 	Protocol          uint8
+
+	// Type is the unix.RTN_* rule type, such as RTN_UNICAST
+	// or RTN_UNREACHABLE.
+	// When adding a new rule, zero means automatic.
+	// Only supported on Linux.
```
Comment (aboch) [inline rule.go:35]: Why does this element need all this explanation while all other ones don't?

Comment (oxtoacart) [reply rule.go:35]: I'm happy to remove it if you prefer, it just seemed like useful documentation for any consumer of this new field.

Comment (aboch) [reply rule.go:35]: Yes let's remove it, unless it deviates from iproute2.
Assumption is who uses this library is familiar with iproute2 and can always look for reference uses there and check kernek code as well. Also you are adding UT which shows an example, which is great.
Thank you for your contribution.

Comment (oxtoacart) [reply rule.go]: Comment removed.

File: rule_linux.go [hunk 2]
```diff
@@ -332,3 +332,34 @@ func ptrEqual(a, b *uint32) bool {
...
 }
+
+func (r Rule) typeString() string {
+	switch r.Type {
+	case unix.RTN_UNSPEC: // zero
+		return ""
+	case unix.RTN_UNICAST:
+		return ""
+	case unix.RTN_LOCAL:
+		return " local"
```
Comment (aboch) [inline rule_linux.go:343]: I am not sure about having the leading space in the string. Let's have the print function add that. Not a big deal if we have a trailing space in the route entry when type == RTN_UNSPEC

File: rule.go [hunk 3]
```diff
@@ -41,8 +42,8 @@ func (r Rule) String() string {
 		to = r.Dst.String()
 	}
 
-	return fmt.Sprintf("ip rule %d: from %s to %s table %d",
-		r.Priority, from, to, r.Table)
+	return fmt.Sprintf("ip rule %d: from %s to %s table %d%s",
```
Comment (aboch) [inline rule.go:45]: I am referring to this stringification of the IP Rule put a space here please

File: rule_linux.go (hunk 2 above)
Comment (oxtoacart) [reply rule_linux.go]: Done. Note that on Mac, we will always have a trailing space.

**Comment by aboch** [comment]
LGTM

**Comment by oxtoacart** [comment]
Thanks for the review @aboch !

**Pull request merged**